src/Agent.Listener/Configuration/WindowsSecurityManager.cs

```diff
@@ -0,0 +1,477 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Cryptography;
+using System.Security.Principal;
+using System.Text;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator (Default = typeof(WindowsSecurityManager))]
+    public interface IWindowsSecurityManager : IAgentService
+    {
+        string GetUniqueBuildGroupName();
+
+        bool LocalGroupExists(string groupName);
+
+        void CreateLocalGroup(string groupName);
+
+        void AddMemberToLocalGroup(string accountName, string groupName);
+
+        void GrantFullControlToGroup(string path, string groupName);
+
+        bool GrantUserLogonAsServicePrivilage(string accountName);
+
+        bool IsValidCredential(string logonAccount, string logonPassword);
+
+        NTAccount GetDefaultServiceAccount();
+
+        void SetPermissionForAccount(string path, string accountName);
+    }
+
+    public class WindowsSecurityManager : AgentService, IWindowsSecurityManager
+    {
+        private const string AgentServiceLocalGroupPrefix = "VSTS_AgentService_G";
+
+        public string GetUniqueBuildGroupName()
+        {
+            return AgentServiceLocalGroupPrefix + GetHashCodeForAgent().Substring(0, 5);
+        }
+
```

Comment:
Specifically, what?

The same general comment applies to WindowsServiceControlManager.cs and AgentService.cs. I see a lot of code, I'm interested to understand what is different about those files from the current Windows agent code.